### PR TITLE
Feat: Login with hostname

### PR DESCRIPTION
**Pull request by ldakhoa**

### CODE REVIEW

File: Sources/APIController/GitHubController+API.swift [hunk 1]
```diff
@@ -19,12 +19,19 @@ extension GitHubController {
...
             }
         }
         
+        private func buildURLString(_ urlString: String) -> String {
+            if UserDefaultManagement.hostname != nil {
+                return "/api/v3\(urlString)"
+            }
+            return urlString
+        }
+
```
Comment (tphduy) [inline Sources/APIController/GitHubController+API.swift:34]: Concats String is easy for a typo bug happens, if possible, lets use the `URL` type to minimize the typo error.
Example:
```Swift
// UserDefaultManagement.hostname: URL?
 case .loginWithToken: 
     return URL(string: "/user", relativeTo: UserDefaultManagement.hostname).absobluteString
```

Comment (tphduy) [reply Sources/APIController/GitHubController+API.swift:34]: An important thing you should know about the relative URL, if your relative URL is actually absolute (having a host itself), then resolving your URL with another base URL (absolute too) will discard the host of the base URL.
Example:
```Swift
// Base URL is ovveriden by relative URL.
// Result is https://lorem.ipsum/foo/bar
print(
    URL(
        string: "https://lorem.ipsum/foo/bar",
        relativeTo: URL(string: "https://foo.bar")!
    )!.absoluteURL
)
```

File: Sources/APIController/NetworkSession+GitHub.swift [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
+        }
+        return makeNetworkSession(fromURL: hostname)
+    }
+    
+    public static var defaultNetworkSession: NetworkSession {
+        UserDefaultManagement.hostname == nil ? github : githubEnterpriseServer
+    }
+    
+    private static func makeNetworkSession(fromURL urlString: String) -> NetworkSession {
+        let baseURL = URL(string: urlString)
```
Comment (tphduy) [inline Sources/APIController/NetworkSession+GitHub.swift:23]: `NetworkSession.github` is enough.

**Review by tphduy — CHANGES_REQUESTED** [review]

File: Sources/APIController/GitHubController+API.swift (hunk 1 above)
Comment (ldakhoa) [reply Sources/APIController/GitHubController+API.swift]: Thank you, fixed in 55cd173

File: Sources/APIController/NetworkSession+GitHub.swift (hunk 2 above)
Comment (ldakhoa) [reply Sources/APIController/NetworkSession+GitHub.swift]: fixed in 55cd173

**Pull request merged**